TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCommon.as

```diff
@@ -0,0 +1,273 @@
+		//print(directions[i].x + " " + directions[i].y);
+        if (isTileSolid(pos + directions[i], map) && !isTileKudzu(test.type)) return 1 + kudzublob; //Can grow while at least 1 solid non kudzu tile in the 8 tiles around it
+    }
+
+	if (Vec2f(0.0f,-8.0f) == dir && this.hasTag("Mut_UpwardLines"))
+	{
+		if (CMap::tile_empty == map.getTile(pos + Vec2f(8.0f, 0.0f)).type && CMap::tile_empty == map.getTile(pos + Vec2f(-8.0f, 0.0f)).type)
+		{
+			return 1 + kudzublob;
+		}
+	}
+
+	if (Vec2f(0.0f,8.0f) == dir && this.hasTag("Mut_DownLines"))
+	{
+		if (CMap::tile_empty == map.getTile(pos + Vec2f(8.0f, 0.0f)).type && CMap::tile_empty == map.getTile(pos + Vec2f(-8.0f, 0.0f)).type)
+		{
+			return 1 + kudzublob;
+		}
+	}
+
+	return 0; //No support found
+	
+}
+
+bool isTileSolid(Vec2f pos, CMap@ map)
+{
+	const u32 offset = map.getTileOffset(pos);
+	if (map.hasTileFlag(offset, Tile::SOLID)) return true;
+	return false;
+}
+
+void MutateTick(CBlob@ this)
+{
+	if (this.hasTag("Mut_Regeneration"))
+	{
+		//print(this.getHealth() + "");
+		this.server_Heal(0.1f);
+	}
+	if (this.hasTag("Mut_Mutating"))
+	{
+		Random@ rand = Random(getGameTime());
+		int r = rand.NextRanged(300);
+		if (r == 0)
+		{
+			Mutate(this);
+		}
+	}
+}
+
+void UpgradeTile(CBlob@ this, Vec2f pos, CMap@ map, Random@ rand)
+{
+	//Create a new core if its time and its chance
+	if (getGameTime() > this.get_u32("Duplication Time") && this.get_u32("Duplication Time") != 0 && rand.NextRanged(30) == 0)
+	{
+		CBlob@ core = server_CreateBlob("kudzucore", 0, pos);
+		if (core != null)
+		{
+			core.getShape().SetStatic(true);
+			Mutate(core); //Offspring start with 1 random mutation
+		}
+		this.set_u32("Duplication Time", 0); //No more duplicating after the first one
+	}
+	else if (this.hasTag("Mut_Badgers") && rand.NextRanged(100) == 0 && getGameTime() > this.get_u32("Upgrade Time"))
+	{
+		CBlob@ node = server_CreateBlob("kudzubadger", 0, pos);
+		if (node != null)
+		{
+			node.getShape().SetStatic(true);
+		}
+		this.set_u32("Upgrade Time", getGameTime() + 900);
+	}
+	else if (this.hasTag("Mut_Explosive") && rand.NextRanged(50) == 0 && getGameTime() > this.get_u32("Upgrade Time"))
+	{
+		CBlob@ node = server_CreateBlob("kudzuexplosive", 0, pos);
+		if (node != null)
+		{
+			node.getShape().SetStatic(true);
+		}
+		this.set_u32("Upgrade Time", getGameTime() + 600);
+	}
+}
+
+void Mutate(CBlob@ this)
+{
+	CParticle@ particle = ParticleAnimated("SmallSmoke", this.getPosition(), Vec2f(0, 0), 0, 1.0f, 2, 0.0f, false);
+	if (particle != null)
+	{
+		particle.Z = 500;
+	}
+
+
+	Random@ rand = Random(getGameTime() + this.getPosition().x); //Randomness is time and position dependent, 
+	//technicly 2 of em in the same coloum mutated at the exact same time would get the same mutation
+	int r = rand.NextRanged(12);
+
+	if (r < 1 && !this.hasTag("Mut_Mutating")) //Possibly the most dangerous mutation, (At first slot to reduce the chance of getting it with other mutations)
```

Comment:
its faster to do == 0 then < 1, nextrange would return an int by default anyhow

you could also not do a hasTag check since if statements are more expensive then just tagging a blob again which does nothing

Reply:
The reason i do this here is so that if it already has a specific mutation rolling that mutation again makes it automagically move onto the next one, thats why its < instead of == and why it checks for the tag

Reply:
ah ok that makes sense